core/trino-main/src/main/java/io/trino/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -254,7 +254,9 @@ private WriterTarget createWriterTarget(WriterTarget target)
         private TableHandle findTableScanHandle(PlanNode node)
         {
             if (node instanceof TableScanNode) {
-                return ((TableScanNode) node).getTable();
+                TableScanNode tableScanNode = (TableScanNode) node;
+                checkState(((TableScanNode) node).isUpdateTarget(), "TableScanNode should be an updatable target");
```

Comment:
nit: `verify` seems more appropriate here
include `node` in the exception message

```suggestion
                verify(((TableScanNode) node).isUpdateTarget(), "TableScanNode should be an updatable target: %s", node);
```